### ISSUE TITLE
Epic: Fix scrolling (as best we can)

### ISSUE THREAD

**Issue by turt2live** [vector-im/element-web#23539]
Q4 (extended quarter) target.

* [ ] POC for new scrolling behaviours
* [ ] Address chronic scroll issues

**Comment by daniellekirkwood** [comment]
Also look into the Jump-to-date feature

**Comment by claell** [comment]
@Johennes Can you give a short update on the course change you mentioned in https://github.com/vector-im/element-web/issues/23540#issuecomment-1420953646? Seems like before it was planned to fix the issues, and now it's an investigation first. What happens if it turns out the issues can't be fixed with reasonable time effort? Leave them as they are?

**Comment by Johennes** [comment]
#23540 meant to prepare a full rewrite of the timeline to fix the various issues in this epic. After reconsidering, we concluded that this isn't a good next step. For one thing, replacing the timeline code requires a mighty investment of effort. For another, most of the issues listed here didn't exist at some point which hints at the possibility to fix the current implementation without a rewrite. The course change is to investigate that latter option. We'll reconsider the subsequent once this is completed – whether successfully or not.

**Comment by claell** [comment]
Sounds good. Though also the description of this issue has been changed from fixing to investigating (and estimating the effort). But in general seems to make sense. As a user, I just hope, this can see some priority, so that https://github.com/matrix-org/matrix-react-sdk/pull/9340 can get merged which might also fix some other stuff as mentioned in https://github.com/matrix-org/matrix-react-sdk/pull/9340#issuecomment-1266128320.
As always, I am available for testing and reproducing, also via video call for example.

**Comment by Johennes** [comment]
> Sounds good. Though also the description of this issue has been changed from fixing to investigating

Yes, we split the two so that we can schedule them separately in the roadmap. So the idea would be to have another epic for the actions that come out of this one.

> I just hope, this can see some priority, so that https://github.com/matrix-org/matrix-react-sdk/pull/9340 can get merged

Yes, https://github.com/matrix-org/matrix-react-sdk/pull/9340 claims to fix https://github.com/vector-im/element-web/issues/23393 which is one of the two tasks under investigation in this epic. So I would expect us to revisit that PR as part of this project. I've added it into the checkbox in the issue description to make this more clear.

**Comment by andybalaam** [comment]
Comment from @MadLittleMods about potentially-related issues. Dropping here in case we want to include them when we work on this:

In addition to timeline scroll problems, there is a timeline reliability problem:

*    Missing thread replies: https://github.com/vector-im/element-web/issues/24466
*    Missing edits: https://github.com/vector-im/element-web/issues/24515
*    Most of the timeline is missing: https://github.com/vector-im/element-web/issues/21432
*    Timeline is completely out of order: https://github.com/vector-im/element-web/issues/21922
*    Timeline is duplicated/looping/repeating:
    *        https://github.com/vector-im/element-web/issues/25195
    *        https://github.com/vector-im/element-web/issues/24354
    *        https://github.com/vector-im/element-web/issues/24676
    *        https://github.com/vector-im/element-web/issues/12471
*    Jumping to unread doesn't work half of the time:
    *        https://github.com/vector-im/element-web/issues/20927
    *        https://github.com/vector-im/element-web/issues/22989
    *        https://github.com/vector-im/element-web/issues/20397
